### ISSUE TITLE
Hosted space doesn't show the published & publicized rooms

### ISSUE THREAD

**Issue by Crementif** [girlbossceo/conduwuit#468]
I'm having some issues related to the room discovery in a space. This might just be my inexperience with Matrix or a budged config, but when I try to join [my server's space](https://matrix.to/#/#cemu:cemu.info) using a matrix.com account, I only see the space but without any of the rooms that I've made public, promoted and published inside of it. Whenever I joined the conduwuit space, I did see the the rooms being promoted, so I think I'm missing something on my side. Maybe it has to due with the rooms being created by the out-of-your-element bot?

The related settings (I think) are set to the following:
```
allow_guest_registration: true
log_guest_registrations = true
allow_guests_auto_join_rooms = true
allow_registration = true
registration_token = "SOME PASSWORD"
allow_federation = true
allow_room_creation = true
block_non_admin_invites = false
allow_public_room_directory_over_federation = true
allow_public_room_directory_without_auth = true
lockdown_public_room_directory = true
allow_device_name_federation = false
```

Thanks for looking into my issue! Let me know if more info is required.

**Comment by GearKite** [comment]
I have the same issue. A temporary fix was adding (non-conduwuit) servers to the [`via`](https://spec.matrix.org/v1.10/client-server-api/#mspacechild) server list ([script for that](https://github.com/GearKite/Matrix-space-via-updater)).

Here's what the spec says about `via`:
> When not present or invalid, the child room is not considered to be part of the space.

Is it possible that clients consider conduwuit to be an invalid server?

**Comment by girlbossceo** [comment]
If you both are running the latest `main` branch, could you give it a try and see if your situation has improved?

**Comment by GearKite** [comment]
Yup, the rooms are visible now, but when trying to join a room with access set to space members this happens: 
`MatrixError: [403] Answer from example.com: [403 / M_FORBIDDEN] M_FORBIDDEN: Event is not authorized.`

**Comment by girlbossceo** [comment]
Okay well at least that's progress. Might be a different issue now and the issue with rooms not showing up are fixed.

Restricted joins via join rules are honestly complex and perhaps that wasn't a case thoroughly tested. I'd make a separate issue about that just so it doesn't get lost in here.

**Comment by girlbossceo** [comment]
I think this is resolved, if it's not please let me know and I'll re-open.

The other join rules issue mentioned is being tracked in a separate issue.
